frontend/src/components/DonorsCard/DonorsForm.tsx

```diff
@@ -0,0 +1,253 @@
+import React, { useEffect, useState } from 'react';
+import './DonorsForm.css';
+import { useAppSelector, useAppDispatch } from '../../app/hooks';
+import { toast } from 'react-toastify';
+
+import {
+  getDonors,
+  updateDonor,
+  deleteDonor,
+  createDonor
+} from '../../features/donors/donorSlice';
+
+
+/* Form to add recipient or donor */
+const DonorsForm = (props: any) => {
+
+  /* Recipient and donor state data here */
+  const [entityType, setEntityType] = useState(props.isUpdate ? props.whichEntity ? props.donor.EntityType : props.recipient.EntityType : '');
```

Comment:
Can you take out all of the propositional logic used to decide recipient/donor? For example whichEntity is used to decide whether to show donor or recipient so it's not necessary anymore. So this and anything along those lines should be removed from the actual form and AdminDataScreen if applicable